### PR TITLE
In case of unauthorized HTTP responses, authorize the client again

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# mender-stress-test-client
+
+[![Build Status](https://gitlab.com/Northern.tech/Mender/mender-stress-test-client/badges/master/pipeline.svg)](https://gitlab.com/Northern.tech/Mender/mender-stress-test-client/pipelines)
```
Comment (lluiscampos) [inline README.md]: This is wrong, should be:
```suggestion
[![Build Status](https://gitlab.com/Northern.tech/Mender/stress-test-client/badges/master/pipeline.svg)](https://gitlab.com/Northern.tech/Mender/stress-test-client/pipelines)
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+the mender-stress-test-client repository. This will put an executable binary in
+the current repo. If a more universal option is wanted, the binary can be
+automatically install into the `~/$GOPATH/bin` repository by running `go install
+.`, and if this is already in your `$PATH` variable, then the stress-test-client
+will function exactly like any other binary in your `PATH`.
+
+### Running
+
+If the client is already installed, run it with the default options like so:
+`mender-stress-test-client --count=<device-count>`
```
Comment (lluiscampos) [inline README.md]: Actually, all this file is using `mender-` prefix, while not in the project.

To be honest, I think there is no point in maintaining the two stress clients. The purpose of `mender-stress-test-client` was scale testing and gui (manual) testing. So I would vote to completely replace the codebase there with this more efficient client, which is the one we are going to use from now on in Load Testing and potentially covers the same needs for gui (manual) testing.

File: client/client.go [hunk 3]
```diff
@@ -162,29 +165,45 @@ func (c *Client) Authenticate() error {
 }
 
 func (c *Client) Run() {
+	inventoryTicker := time.NewTicker(c.Config.InventoryInterval)
+	updateTicker := time.NewTicker(c.Config.UpdateInterval)
+
+auth:
```
Comment (lluiscampos) [inline client/client.go:171]: This is the very first time I se `goto` in golang :exploding_head:

**Review by lluiscampos — CHANGES_REQUESTED** [review]
The code looks good. The README deserves a discussion, see below :)

**Comment by tranchitella** [comment]
@lluiscampos this was more a request for comments, indeed. I don't know how to proceed: shall we move the code back to the `mender-stress-test-client`, replacing that implementation? Or spin-off a new stress test client with a different name?

**Comment by tranchitella** [comment]
I removed the commit which introduces the README.md, until we decide how to proceed.

**Comment by lluiscampos** [comment]
> @lluiscampos this was more a request for comments, indeed. I don't know how to proceed: shall we move the code back to the `mender-stress-test-client`, replacing that implementation? Or spin-off a new stress test client with a different name?

I think we can review it and replace it. I'll create a task for it in the context of the Scale Testing, and make sure I get the feedback from all relevant people.

**Review by lluiscampos — APPROVED** [review]

File: client/client.go (hunk 3 above)
Comment (tranchitella) [reply client/client.go:171]: it was the cleanest way to "restart" the client without making the code more complicated

**Pull request merged**